src/serialization/UntypedNumber.cs

```diff
@@ -0,0 +1,18 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Kiota.Abstractions.Serialization
+{
+    /// <summary>
+    /// Represents an untyped node with number value.
+    /// </summary>
+    /// <param name="value">The number value associated with the node.</param>
+    public class UntypedNumber(string value) : UntypedNode
+    {
+        /// <summary>
+        /// The number associated with untyped number node.
+        /// </summary>
+        public new string Value { get; } = value;
```

Comment:
something I missed at the beginning of the review, any specific reason why this is a string and not one of the number types?

Reply:
At the beginning, I wanted to create specific types like UntypedFloat, UntypedDouble, UntypedLong, UntypedInteger, etc. but without the context I don't know what's the correct type.
Same for UntypedString where the `Value` can represent string, Guid, DateTime or DateTimeOffset.
If it makes sense, I can split UntypedNumber to UntypedDecimal, UntypedDouble, UntypedFloat, UntypedLong and UntypedInteger and split UntypedString to UntypedString, UntypedDateTime, UntypedDateTimeOffset and UntypedGuid

Reply:
I've split UntypedNumber to UntypedDecimal, UntypedDouble, UntypedFloat, UntypedLong and UntypedInteger.
Draft for serialization updated: https://github.com/microsoft/kiota-serialization-json-dotnet/pull/167